sfyra/pkg/tests/server.go

```diff
@@ -260,6 +260,41 @@ func TestServerAcceptance(ctx context.Context, metalClient client.Client, vmSet
 	}
 }
 
+// TestServerResetOnAcceptance tests that servers are reset when accepted.
+func TestServerResetOnAcceptance(ctx context.Context, metalClient client.Client, vmSet *vm.Set) TestFunc {
+	return func(t *testing.T) {
+		servers := &v1alpha1.ServerList{}
+		err := metalClient.List(ctx, servers)
+		require.NoError(t, err)
+
+		for _, server := range servers.Items {
+			server := server
+
+			patchHelper, err := patch.NewHelper(&server, metalClient)
+			require.NoError(t, err)
+
+			server.Spec.Accepted = true
```

Comment:
we should count number of servers transitioned into `Accepted` here, and in the next `retry` block test only those servers to get things idempotent

Reply:
I'm not sure I understand fully. I will try and you can tell me if I got it right.